stickynotes/stickynotes.c

```diff
@@ -50,978 +50,1055 @@ set_icon_geometry (GdkWindow *window,
-	if (!color_str || force_default)
-	{
-		if (use_system_color)
-			color_str_actual = NULL;
-		else
-			color_str_actual = g_settings_get_string (stickynotes->settings, "default-color");
-	}
-	else
-		color_str_actual = g_strdup (color_str);
-
-	if (!font_color_str || force_default)
-	{
-		if (use_system_color)
-			font_color_str_actual = NULL;
-		else
-			font_color_str_actual = g_settings_get_string (stickynotes->settings, "default-font-color");
-	}
-	else
-		font_color_str_actual = g_strdup (font_color_str);
-
-	/* Do not use custom colors if "use_system_color" is enabled */
-	if (color_str_actual) {
-		GdkRGBA colors[4];
-		gint i;
-
-		for (i = 0; i <= 3; i++)
-		{
-			gdk_rgba_parse (&colors[i], color_str_actual);
-			colors[i].red = (colors[i].red * (10 - i)) / 10;
-			colors[i].green = (colors[i].green * (10 - i)) / 10;
-			colors[i].blue = (colors[i].blue * (10 - i)) / 10;
-		}
-
-		gtk_widget_override_background_color (note->w_window, GTK_STATE_FLAG_NORMAL, &colors[0]);
-		gtk_widget_override_background_color (note->w_body, GTK_STATE_FLAG_NORMAL, &colors[0]);
-		gtk_widget_override_background_color (note->w_lock, GTK_STATE_FLAG_NORMAL, &colors[0]);
-		gtk_widget_override_background_color (note->w_close, GTK_STATE_FLAG_NORMAL, &colors[0]);
-		gtk_widget_override_background_color (note->w_resize_se, GTK_STATE_FLAG_NORMAL, &colors[0]);
-		gtk_widget_override_background_color (note->w_resize_sw, GTK_STATE_FLAG_NORMAL, &colors[0]);
-	} else {
-		gtk_widget_override_background_color (note->w_window, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_background_color (note->w_body, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_background_color (note->w_lock, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_background_color (note->w_close, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_background_color (note->w_resize_se, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_background_color (note->w_resize_sw, GTK_STATE_FLAG_NORMAL, NULL);
-	}
-
-	if (font_color_str_actual)
-	{
-		GdkRGBA color;
-
-		gdk_rgba_parse (&color, font_color_str_actual);
-
-		gtk_widget_override_color (note->w_window, GTK_STATE_FLAG_NORMAL, &color);
-		gtk_widget_override_color (note->w_body, GTK_STATE_FLAG_NORMAL, &color);
-	}
-	else
-	{
-		gtk_widget_override_color (note->w_window, GTK_STATE_FLAG_NORMAL, NULL);
-		gtk_widget_override_color (note->w_body, GTK_STATE_FLAG_NORMAL, NULL);
-	}
-
-	if (color_str_actual)
-		g_free (color_str_actual);
-	if (font_color_str_actual)
-		g_free (font_color_str_actual);
+    char *color_str_actual, *font_color_str_actual;
+    gboolean force_default, use_system_color;
+
+    if (save) {
+        if (note->color)
+            g_free (note->color);
+        if (note->font_color)
+            g_free (note->font_color);
+
+        note->color = color_str ?
+            g_strdup (color_str) : NULL;
+        note->font_color = font_color_str ?
+            g_strdup (font_color_str) : NULL;
```

Comment:
no need to break the ternary operator statement, it can be placed in one line:
```
note->color = color_str ? g_strdup (color_str) : NULL;
note->font_color = font_color_str ? g_strdup (font_color_str) : NULL;
```